src/data/move-attrs/move-attr.ts

```diff
@@ -11,9 +20,41 @@ import type { MoveCondition, MoveConditionFunc } from "../move-conditions";
 export abstract class MoveAttr {
   /** Should this {@linkcode Move} target the user? */
   public selfTarget: boolean;
+  protected options?: MoveAttrOptions;
 
-  constructor(selfTarget: boolean = false) {
+  constructor(selfTarget: boolean = false, options?: MoveAttrOptions) {
     this.selfTarget = selfTarget;
+    this.options = options;
+  }
+
+  /**
+   * Defines whether or not this attribute contributes to effect score even when the move
+   * KOs its target.
+   * @default false
+   * @see {@linkcode getEffectScore}
+   */
+  public get appliesScoreOnKO(): boolean {
+    return this.options?.appliesScoreOnKO ?? false;
```

Comment:
Are the `?? false` needed? Could set all the variables to be false by default

Reply:
Not supported for interfaces:

![image](https://github.com/user-attachments/assets/d05a426f-2275-4ffb-a752-9bc156efbfb2)

Just following the conventions for other `options` interfaces in the attributes. Putting default values in the constructor also seems awkward imo

Reply:
```suggestion
    return !!this.options?.appliesScoreOnKO;
```
You could do this instead (same for the others).

Reply:
I like it as is a bit more because the default values are separated and can easily be tweaked.